### PR TITLE
Add  Rover brake test

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by rmackay9** [comment]
I'm happy with this test going in.  I've got a fix by the way to the rover braking issue that we've seen (the motors could go slightly negative even though ATC_BRAKE was 0.
https://github.com/rmackay9/rmackay9-ardupilot/commit/b846e7735268e36524c3acaf88f838fc78107503

File: Tools/autotest/apmrover2.py [hunk 1]
```diff
@@ -122,6 +122,65 @@ def do_set_mode_via_command_long(mavproxy, mav):
...
+
+    set_parameter(mavproxy, 'CRUISE_SPEED', old_cruise_speed)
+    set_parameter(mavproxy, 'ATC_ACCEL_MAX', old_accel_max)
+
+    return delta
+
+def drive_brake(mavproxy, mav):
+    old_using_brake = get_parameter(mavproxy, 'ATC_BRAKE')
+
+    set_parameter(mavproxy, 'CRUISE_SPEED', 15)
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply Tools/autotest/apmrover2.py]: Will fix.

File: Tools/autotest/apmrover2.py [hunk 2]
```diff
@@ -122,6 +122,65 @@ def do_set_mode_via_command_long(mavproxy, mav):
...
+
+    set_parameter(mavproxy, 'ATC_BRAKE', 0)
+    distance_without_brakes = drive_brake_get_stopping_distance(mavproxy, mav, 15)
+
+    # brakes on:
+    set_parameter(mavproxy, 'ATC_BRAKE', 1)
+    distance_with_brakes = drive_brake_get_stopping_distance(mavproxy, mav, 15)
+    # brakes off:
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply Tools/autotest/apmrover2.py]: Will fix.

File: Tools/autotest/apmrover2.py [hunk 3]
```diff
@@ -122,6 +122,65 @@ def do_set_mode_via_command_long(mavproxy, mav):
...
+    set_parameter(mavproxy, 'ATC_BRAKE', old_using_brake)
+
+    delta = distance_with_brakes - distance_without_brakes
+    print("with=%0.2f without=%0.2f delta=%0.2f\n" % (distance_with_brakes, distance_without_brakes, delta),)
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply Tools/autotest/apmrover2.py]: Will remove.

File: Tools/autotest/apmrover2.py [hunk 4]
```diff
@@ -122,6 +122,65 @@ def do_set_mode_via_command_long(mavproxy, mav):
...
+    if delta > 0 or abs(delta) < distance_without_brakes*0.05: # 5% isn't asking for much
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply Tools/autotest/apmrover2.py]: Nice.

**Comment by peterbarker** [comment]
I've pushed this in.

Thanks @OXINARF !

**Pull request closed without merging**